### ISSUE TITLE
stdout is truncated at 512 characters

### ISSUE THREAD

**Issue by bennuttall** [mu-editor/mu#707]
I've found that and stdout is limited to 512 characters.

```python
print('|'*511 + 'x')
```

works as expected. I see 511 pipes followed by `x`:

```
|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||x
```

However:

```python
print('|'*512 + 'x')
```

I only see 512 pipes, no `x`:

```
||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
```

This seems to be the cause of the pgzero issue I opened earlier: #706.

**Comment by tjguk** [comment]
Thanks, Ben. This seems to be a popular issue today -- and there's a PR #685 currently waiting to be tested to fix this

**Comment by ntoll** [comment]
Fixed by #723
